rs/orchestrator/src/firewall.rs

```diff
@@ -262,38 +262,64 @@ impl Firewall {
         // in the registry inclusive.
         let registry_versions = self.get_registry_versions(registry_version);
 
-        // Get the IPs of all nodes on system subnets
-        let system_subnet_node_ips: BTreeSet<IpAddr> = registry_versions
-            .into_iter()
-            .flat_map(|registry_version| {
-                self.registry
-                    .get_system_subnet_nodes_ip_addresses(registry_version)
-                    .inspect_err(|err| {
-                        warn!(
-                        every_n_seconds => 30,
-                        self.logger,
-                        "Failed to get the IPs of system subnet nodes in the registry: {}", err)
-                    })
-                    .unwrap_or_default()
-            })
-            .collect();
+        // Depending on whether the node is a system or app API boundary node, get the IPs of all
+        // system subnet or application subnet nodes
+        let node_ips: BTreeSet<IpAddr> = match self
+            .registry
+            .is_system_api_boundary_node(self.node_id, registry_version)
+        {
+            Ok(true) => registry_versions
+                .into_iter()
+                .flat_map(|registry_version| {
+                    self.registry
+                        .get_system_subnet_nodes_ip_addresses(registry_version)
+                        .inspect_err(|err| {
+                            warn!(
+                            every_n_seconds => 30,
+                            self.logger,
+                            "Failed to get the IPs of system subnet nodes in the registry: {}", err)
```

Comment:
nit: I think there is something wrong with the formatting here?  It might be because the string is too long (`rustfmt` sometimes skips formatting when it encounters too long of a string)

```suggestion
                            warn!(
                            every_n_seconds => 30,
                            self.logger,
                            "Failed to get the IPs of system subnet \
                            nodes in the registry: {}", err)
```